### ISSUE TITLE
Set unique bluetooth device public-address if none is set

### ISSUE THREAD

**Issue by xnox** [systemd/systemd#29346]
### Component

systemd-udevd

### Is your feature request related to a problem? Please describe

I am working on a device that has bad drivers. Bluetooth is available but not usable as MAC is not set by the kernel. After setting public-address with btmgr to anything, it now works.

### Describe the solution you'd like

I wish for udev to be able to set mac address on bluetooth devices, by deriving a random one based on machine-id & bluetooth device name, such that it is unique & persistent. Similar to what udev already can do for network devices.

### Describe alternatives you've considered

Currently I have userspace systemd service to set mac address.

### The systemd version you checked that didn't have the feature you are asking for

253

**Comment by xnox** [comment]
> but doesn't bluetooth have a userspace daemon anyway (bluez?) which can do this much better than we could?

will check, for me bluez seems to do nothing, and expects someone to set the address. Will dig more to see if it already can do this, or be told to do it.

**Comment by xnox** [comment]
ps. i hate broken hardware drivers for broken chipsets =/

**Comment by erebion** [comment]
> will check, for me bluez seems to do nothing, and expects someone to set the address. Will dig more to see if it already can do this, or be told to do it.

Have you been able to comu up with anything?

This is currently the main frustration on my phone, which runs Debian. After every reboot I have to manually set a MAC address in order to use Bluetooth:

```
echo $"stdin for btmgmt" | btmgmt -i hci0 public-addr $(hciconfig hci0 | grep -o $"[[:xdigit:]:]\\{11,17\\}")
```

(btmgmt does not run correctly if there's no input, so I have a script which looks weird, lol)

**Comment by xnox** [comment]
yes i too came up with weird scripts / systemd unit calling weird script.

**Comment by erebion** [comment]
> yes i too came up with weird scripts / systemd unit calling weird script.

Please do post your solution somewhere. I still have not found anything that works well.
